source/dubregistry/dbcontroller.d

```diff
@@ -177,6 +177,35 @@ class DbController {
 		m_packages.update(["name": packname], ["$set": ["repository": repo]]);
 	}
 
+	void setPackageLogo(string packname, bdata_t png)
+	{
+		import std.digest.md : md5Of;
+
+		if (png.length)
+		{
+			m_packages.update(["name": packname], ["$set": [
+				"logo": BsonBinData(BsonBinData.Type.generic, png),
+				"logoHash": BsonBinData(BsonBinData.Type.md5, cast(bdata_t)md5Of(png)[].idup)
+			]]);
+		}
+		else
+			m_packages.update(["name": packname], ["$unset": ["logo": 1, "logoHash": 1]]);
```

Comment:
It's quite untypical to use `1` as null value.

Reply:
this is `$unset`, not `$set`

Reply:
Yes and I don't mind that you use `1`.
It's just more common to use `0` or `-1`

(Of course that mongodb itself promotes such a crap API is a different story)

Reply:
hm just took the first result off google when searching for how to unset